deepspeed/runtime/zero/stage3.py

```diff
@@ -2984,42 +2867,49 @@ def backward(self, loss, retain_graph=False):
         2. scaled_loss = fp32_loss*loss_scale
         3. scaled_loss.backward(), which accumulates scaled gradients into the ``.grad`` attributes of the model's fp16 leaves
         """
-        self.micro_step_id += 1
-        print_rank_0(
-            f"Total fully available parameters {self.param_coordinator.total_available_parameter_numel}"
-        )
-
         if self.swap_optimizer:
             self.optimizer_swapper.pre_backward()
 
         see_memory_usage(f"Before backward", force=False)
-        if self.contiguous_gradients:
```

Comment:
@jfc4050, it seems the contiguous_gradients and overlap_communication logic is removed, Is that correct?

Reply:
contiguous gradients and communication overlap are still supported, its just that this piece of code doesnt need to be aware of either anymore.

the IPG buffer (there's one of them now rather than two) is now allocated once during initialization and kept for the length of training